Source/Core/Core/IOS/USB/Emulated/Skylander.cpp

```diff
@@ -906,12 +907,19 @@ int SkylanderUSB::SubmitTransfer(std::unique_ptr<IntrMessage> cmd)
   // echo the request as the response and respond after 1ms
   if (cmd->length > 32 && cmd->length <= 64)
   {
+    // Play audio through Portal Mixer
+    // Audio is unsigned 16 bit, supplied as 64 bytes which is 32 samples
+    SoundStream* sound_stream = system.GetSoundStream();
+    void* address = memory.GetPointer(cmd->data_address);
+    sound_stream->GetMixer()->PushSkylanderPortalSamples(reinterpret_cast<short*>(address), 32);
```

Comment:
This `32` only works if the `cmd->length == 64`. Either only push if that's actually the case or make it work if it's shorter than that too.

The cast to short here is also not great. Might be worth rewriting `PushSkylanderPortalSamples()` to take a `const u8*` instead, or locally memcpy to a `std::array<s16>`.